### PR TITLE
Automatically Populate New Visual Groups

### PR DESCRIPTION
**Pull request by mo-nathan**
Made it so creating a VisualGroup automatically includes a bunch of images (confidence > 1.5).  This works reasonably for adding single names.

I also added a way to add larger lists of names.  Ended up implementing it as a rake task.  It was able to create Alan Celestino's list of 1246 names in just under 12 minutes.

### CODE REVIEW

File: app/models/visual_group.rb [hunk 1]
```diff
@@ -38,4 +35,13 @@ def distinct_names
...
     visual_group_images.where(included: true).pluck(:image_id)
   end
+
+  private
+
+  def image_ids_by_name_vote(name, vote)
+    ObservationImage.joins({ observation: :name }).where(
+      "observations.vote_cache >= ? AND names.text_name = ?", vote, name
+    ).order("observations.vote_cache desc").
+      pluck(:image_id, "observations.vote_cache")
```
Comment (JoeCohen) [inline app/models/visual_group.rb:45]:
```suggestion
    ObservationImage.joins({ observation: :name }).
      where(Observation[:vote_cache].gt(vote)).
      where(names: { text_name: name }).
      order(vote_cache: :desc).
      pluck(:image_id, :vote_cache)
```

File: app/views/shared/_visual_group_table.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      </thead>
+      
+      <tbody>
+        <% included_counts = visual_model.image_counts(true)
+           excluded_counts = visual_model.image_counts(false)
+           @visual_model.visual_groups.order(:name).each do |visual_group| %>
+          <tr>
+            <td><%= link_to(visual_group.name, visual_group) %></td>
+      	<td><%= included_counts[visual_group.id] || 0 %>
+      	<td><%= excluded_counts[visual_group.id] || 0 %>
```
Comment (JoeCohen) [inline app/views/shared/_visual_group_table.html.erb:34]:
```suggestion
      	    <td><%= included_counts[visual_group.id] || 0 %></td>
      	    <td><%= excluded_counts[visual_group.id] || 0 %></td>
```

**Comment by mo-nathan** [comment]
The "test (push)" failure says:

Progress: |===ERROR Admin::BlockedIpsControllerTest#test_blocked_ips (10.08s)
Minitest::UnexpectedError:         Errno::ENOENT: No such file or directory @ rb_sysopen - /home/runner/work/mushroom-observer/mushroom-observer/config/blocked_ips.txt
            app/classes/ip_stats.rb:204:in `initialize'
            app/classes/ip_stats.rb:204:in `open'
            app/classes/ip_stats.rb:204:in `rewrite_file'
            app/classes/ip_stats.rb:183:in `rewrite_blocked_ips'
            app/classes/ip_stats.rb:[11](https://github.com/MushroomObserver/mushroom-observer/actions/runs/3933868940/jobs/6728029675#step:13:12)2:in `remove_blocked_ips'
            test/controllers/admin/blocked_ips_controller_test.rb:9:in `test_blocked_ips'

========================================================grep: /home/runner/work/mushroom-observer/mushroom-observer/log/test.log: No such file or directory
=========|

I see we have gotten this error from time to time in the past.  Is this a symptom of something specific and known or is it an intermittent failure?

**Comment by nimmolo** [comment]
@mo-nathan That has been an intermittent failure. I think @JoeCohen or @pellaea know more about it. 

I've never figured it out.

**Comment by pellaea** [comment]
I don't recall, but it would seem easy enough simply to create an empty
file:

mo> touch config/blocked_ips.txt

Right?  Does that work?  Or is this the automated tester on github?  If so,
maybe we should have the setup script ensure the file exists?  Maybe it's
intermittent because most queries in controller tests will automatically
create it if it doesn't exist, but apparently there are at least some tests
which require it to exist before it has been automatically created in
another test.

**Comment by JoeCohen** [comment]
I don't recall seeing this. I don't know what it's about.

**Comment by mo-nathan** [comment]
I'm not going to sweat the blocked_ips.txt file thing then for this PR.  If/when it happens again, grabbing the seed for the tests should allow us to reproduce it.

**Comment by pellaea** [comment]
Good idea!

**Pull request merged**